crowbar_framework/lib/crowbar/checks/network.rb

```diff
@@ -0,0 +1,128 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "resolv"
+require "ipaddr"
+require "socket"
+require "logger"
+
+module Crowbar
+  module Checks
+    class Network
+      attr_accessor :logger
+
+      class << self
+        def ip_addrs(fqdn, version = :ipv4)
+          ip_addrs = []
+
+          Resolv.getaddresses(fqdn).each do |address|
+            ip_addr = IPAddr.new(address)
+            if version == :ipv6
+              ip_addrs.push address if ip_addr.ipv6?
+            else
+              ip_addrs.push address if ip_addr.ipv4?
+            end
+          end
+
+          ip_addrs
+        end
+      end
+
+      def initialize
+      end
+
+      def fqdn_detected?
+        return false if hostname.blank? || domain.blank? || fqdn.blank?
+
+        true
+      end
+
+      def ip_resolved?
+        return false if ipv4_addrs.empty? && ipv6_addrs.empty?
+
+        true
+      end
+
+      def loopback_resolved?
+        return false if ipv4_addrs.find { |e| /^127/ =~ e } || ipv6_addrs.find { |e| /^::[0-9]$/ =~ e }
+
+        true
+      end
+
+      def ip_configured?
+        ipv4_configured = false
+        ipv6_configured = false
+
+        addr_infos = Socket.ip_address_list
+        addr_infos.each do |addr_info|
+          ipv4_configured = true if ipv4_addrs.include?(addr_info.ip_address)
+          ipv6_configured = true if ipv6_addrs.include?(addr_info.ip_address)
+        end
+
+        return false unless ipv4_configured || ipv4_addrs.empty?
+        return false unless ipv6_configured || ipv6_addrs.empty?
+
+        true
+      end
+
+      def network_valid?
+        if File.exist?("/etc/crowbar/network.json")
+          network_json = "/etc/crowbar/network.json"
+        else
+          network_json = Rails.root.join("..", "chef", "data_bags", "crowbar", "template-network.json").to_s
```

Comment:
Metrics/LineLength: Line is too long. [108/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)